### PR TITLE
Fixed costs form

### PR DESCRIPTION
**Pull request by isabellelii**
#### Allows user to add monthly fixed expenses 
Changes proposed in this pull request:

* add fixed expenses form

#### What I have learned working on this feature: 
* Adding migrations to change existing databases 
* How to use form_with 
* When to use controllers and how do use them

### CODE REVIEW

File: app/views/home/index.html.erb [hunk 1]
```diff
@@ -15,3 +15,5 @@
 <% flash.each do |key, value| %>
   <div class="alert alert-<%= key %>"><%= value %></div>
 <% end %>
+
+<%= link_to "Fixed Costs", new_expense_path %>
```
Comment (magnus-thor) [inline app/views/home/index.html.erb]: Change this link to `New Expense'

File: config/routes.rb [hunk 2]
```diff
@@ -5,7 +5,9 @@
   resources :incomes, only: [:new, :create]
 
   devise_for :users
-  
+
+  resources :expenses
+
   root controller: :home, action: :index
 
   root 'home#index'
```
Comment (oliverochman) [inline config/routes.rb]: ```root 'home#index'```

this is not needed when you have the line of code above

**Review by AKidd95 — COMMENTED** [review]
``` 
Then("I fill in {string} field with {string}") do |input, value|
  fill_in(input, :with => value)
end 
```

Old syntax in fill_in line

File: spec/models/expense_spec.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+require 'rails_helper'
+
```
Comment (oliverochman) [inline spec/models/expense_spec.rb:2]: you need to test Factorybot validation

File: features/fixed_costs_form.feature [hunk 4]
```diff
@@ -0,0 +1,17 @@
+Feature: Be able to enter monthly fixed costs
+As a user
+In order to know what my fixed cost total is
+I should be able to enter all the fields
```
Comment (magnus-thor) [inline features/fixed_costs_form.feature]: Fix the indentation

File: features/fixed_costs_form.feature [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  When I click on "New Expense"
+  And I should see "Enter Fixed Expenses"
+  And I fill in 'Rent' field with '30000'
+  And I fill in 'Insurance' field with '300'
+  And I fill in 'Transportation' field with '1000'
+  And I fill in 'Education' field with '380'
+  And I fill in 'Food' field with '3500'
+  And I fill in 'Other' field with '2000'
+  And I click on "Submit"
+  And I should see "New Income Expense saved!"
```
Comment (magnus-thor) [inline features/fixed_costs_form.feature]: change all double qoutes to single qoutes

**Review by magnus-thor — APPROVED** [review]
Looks good.

**Pull request merged**